sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayAsyncClient.java

```diff
@@ -0,0 +1,82 @@
+import com.azure.communication.networktraversal.models.CommunicationRelayConfigurationRequest;
+import com.azure.communication.networktraversal.models.CommunicationErrorResponseException;
+import com.azure.communication.common.CommunicationUserIdentifier;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import reactor.core.publisher.Mono;
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Asynchronous client interface for Azure Communication Network Traversal
+ * operations
+ */
+@ServiceClient(builder = CommunicationRelayClientBuilder.class, isAsync = true)
+public final class CommunicationRelayAsyncClient {
+
+    private final CommunicationNetworkTraversalsImpl client;
+    private final ClientLogger logger = new ClientLogger(CommunicationRelayAsyncClient.class);
+
+    CommunicationRelayAsyncClient(CommunicationNetworkingClientImpl communicationNetworkingClient) {
+        client = communicationNetworkingClient.getCommunicationNetworkTraversals();
+    }
+
+    /**
+     * Creates a new CommunicationRelayConfiguration.
+     * 
+     * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @return The created Communication Relay Configuration.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<CommunicationRelayConfiguration> getRelayConfiguration(CommunicationUserIdentifier communicationUser) {
+        CommunicationRelayConfigurationRequest body = new CommunicationRelayConfigurationRequest();
+        body.setId(communicationUser.getId());
+        
+        return Mono.just(client.issueRelayConfiguration(body));
+    }
+
+    /**
+     * Creates a new CommunicationRelayConfiguration with response.
+     *
+     * @param communicationUser The CommunicationUserIdentifier for whom to issue a token
+     * @return The created Communication Relay Configuration.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CommunicationRelayConfiguration>> getRelayConfigurationWithResponse(CommunicationUserIdentifier communicationUser) {
+        try {
+            CommunicationRelayConfigurationRequest body = new CommunicationRelayConfigurationRequest();
+            body.setId(communicationUser.getId());
+            return client.issueRelayConfigurationWithResponseAsync(body)
+                .onErrorMap(CommunicationErrorResponseException.class, e -> translateException(e))
+                .flatMap(
+                    (Response<CommunicationRelayConfiguration> response) -> {
+                        return Mono.just(
+                            new SimpleResponse<CommunicationRelayConfiguration>(
+                                response,
+                                response.getValue()));
```

Comment:
Lines 63-68 is usually to convert the response value to another type and putting it into Response<>. I don't see that needed here. I think you can just remove these lines here.

Reply:
Yeah, I think it should have worked since they are the same type, but the compiler complains. Seems I have to keep it this way. I guess that's why in identity is done the same way.

Reply:
I think you should look into why the compiler is complaining because you should not need this block of code. 
Identity uses a flatmap because it needs to manipulate the response type, that is not the case here